File: src/modul5/dataMahasiswa.java
package modul5;

import java.io.*;
import java.util.Scanner;

public class dataMahasiswa {

  static dataMahasiswa obj = new dataMahasiswa();
  static Scanner sc = new Scanner(System.in);

  public static void main(String[] args) {
    obj.user();
  }

  public void user() {
    int x = 0;
    System.out.println("\nProgram input data mahasiswa");
    System.out.println("1. Read\n2. Insert\n3. Delete");
    System.out.print("Input pilihan: ");
    x = sc.nextInt();

    if (x == 1) {
      obj.read();
    } else if (x == 2) {
      obj.insert();
    } else if (x == 3) {
      obj.delete();
    } else {
      System.exit(1);
    }
  }

  public void read() {
    try {
      File file = new File("data_mahasiswa.txt");
      Scanner dr = new Scanner(file);
      while (dr.hasNextLine()) {
        String fileData = dr.nextLine();
        System.out.println(fileData);
      }
      dr.close();
      obj.user();
    } catch (Exception e) {
      System.out.println("Exception Occurred");
    }
  }

  public void insert() {
    try {
      FileWriter fw = new FileWriter("data_mahasiswa.txt", true);

      String nama, nim, sem;

      String text = "";

      while (!text.equals("n")) {
        System.out.print("Input Nama: ");
        nama = sc.next();
        fw.write("\n" + nama + ", ");

        System.out.print("Input NIM: ");
        nim = sc.next();
        fw.write(nim + ", ");

        System.out.print("Input Semester: ");
        sem = sc.next();
        fw.write(sem);

        System.out.print("Continue? y/n: ");
        text = sc.next();
      }
      fw.close();
      System.out.println("\nFile berhasil diupdate");
      obj.user();
    } catch (Exception e) {
      System.out.println("Exception Occured");
    }
  }

  public void delete() {
    try {
      File inputFile = new File("data_mahasiswa.txt");
      File tempFile = new File("temp.txt");

      BufferedReader br = new BufferedReader(new FileReader(inputFile));
      PrintWriter pw = new PrintWriter(new FileWriter(tempFile));

  
      String text = "";
      int line = 0, deleteLine = 0;

      while (!text.equals("n")) {
        System.out.print("Pilih baris yang ingin di hapus:");
        deleteLine = sc.nextInt();
        String currentLine;
        while ((currentLine = br.readLine()) != null) {
          line++;
          if (deleteLine != line) {
            pw.println(currentLine);
          }
        }
        pw.close();
        br.close();
        inputFile.delete();
        tempFile.renameTo(inputFile);
        System.out.println("Data berhasil dihapus.");

        System.out.print("Continue? y/n: ");
        text = sc.next();
      }
      obj.user();
    } catch (Exception e) {
      System.out.println("Exception Occured");
    }
  }
  


}

File: bin/modul5/datamahasiswa.py
def user():
    try:
        print("\nProgram input data mahasiswa")
        print("1. Tambah Data\n2. Tampilkan Data\n3. Update Data\n4. Delete Data\n5. Search Data\n6. Keluar")

        x = int(input("Input pilihan: "))

        if x == 1:
            insert()
        elif x == 2:
            read()
        elif x == 3:
            update()
        elif x == 4:
            delete()
        elif x == 5:
            search()
        else:
            exit(1)
    except ValueError:
        print("Masukkan angka yang valid!")


def insert():
    file = open("mahasiswa.txt", "a")
    nama = input("Masukkan Nama: ")
    nim = input("Masukkan NIM: ")
    matkul = input("Masukkan Mata Kuliah: ")
    sem = input("Input semester: ")
    file.write(nama+", "+nim+", "+matkul+", "+sem+"\n")
    file.close()
    print('Data berhasil ditambahkan')
    user()


def read():
    file = open("mahasiswa.txt", "r")
    i = 0
    for line in file:
        i += 1

        print("Menampilkan Data ke-", i, ":")
        print("Nama:", line.split(', ')[0])
        print("NIM:", line.split(', ')[1])
        print("Mata Kuliah:", line.split(', ')[2])
        print("Semester:", line.split(', ')[3])
    file.close()
    user()


def update():

    newNim = input("\nMasukkan NIM data yang ingin diupdate: ")
    with open("mahasiswa.txt", "r+") as file:
        lines = file.readlines()
        for index, line in enumerate(lines):
            if newNim in line:
                nama = input("Input Nama: ")
                nim = input("Input NIM: ")
                matkul = input("Masukkan Mata Kuliah: ")
                sem = input("Input semester: ")

                updated_line = f"{nama}, {nim},{matkul}, {sem}\n"
                lines[index] = updated_line

        file.seek(0)
        file.truncate()

        file.writelines(lines)

    print('Data berhasil di update')
    user()


def delete():
    file = open("mahasiswa.txt", "r")
    i = 0
    for line in file:
        i += 1
        print("Menampilkan Data ke-", i, ":")
        print("Nama:", line.split(', ')[0])
        print("NIM:", line.split(', ')[1])
        print("Mata Kuliah:", line.split(', ')[2])
        print("Semester:", line.split(', ')[3])
    file.close()

    nim = input("\nMasukkan NIM data yang ingin dihapus: ")

    with open("mahasiswa.txt", "r") as file:
        lines = file.readlines()
    with open("mahasiswa.txt", "w") as file:

        for line in lines:
            if nim not in line:
                file.write(line)

    print('Data berhasil di hapus')
    user()


def search():

    search = input("\nMasukkan NIM data yang ingin dicari: ")
    with open("mahasiswa.txt", "r") as file:

        lines = file.readlines()
        for line in lines:
            if line.find(search) != -1:
                print("Nama: ", line.split(', ')[0])
                print("NIM: ", line.split(', ')[1])
                print("Mata Kuliah: ", line.split(', ')[2])
                print("Semester: ", line.split(', ') [3])

    user()


user()
